examples/pytorch/vision/run_image_classification.py

```diff
@@ -0,0 +1,313 @@
+
+    if data_args.validation_dir is not None:
+        eval_dataset = ImageFolder(
+            data_args.validation_dir,
+            T.Compose(
+                [
+                    T.Resize(data_args.image_size),
+                    T.CenterCrop(data_args.image_size),
+                    T.ToTensor(),
+                    normalize,
+                ]
+            ),
+        )
+    else:
+        indices = torch.randperm(len(train_dataset)).tolist()
+        n_val = math.floor(len(indices) * data_args.train_val_split)
+        train_dataset = torch.utils.data.Subset(train_dataset, indices[:-n_val])
+        eval_dataset = torch.utils.data.Subset(train_dataset, indices[-n_val:])
+
+    test_dataset = (
+        ImageFolder(
+            data_args.test_dir,
+            T.Compose(
+                [
+                    T.Resize(data_args.image_size),
+                    T.CenterCrop(data_args.image_size),
+                    T.ToTensor(),
+                    normalize,
+                ]
+            ),
+        )
+        if data_args.test_dir is not None
+        else None
+    )
+
+    metric = datasets.load_metric("accuracy")
+
+    def compute_metrics(p):
+        return metric.compute(predictions=np.argmax(p.predictions, axis=1), references=p.label_ids)
+
+    config = AutoConfig.from_pretrained(
+        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+        num_labels=len(labels),
+        label2id=label2id,
+        id2label=id2label,
+        finetuning_task="image-classification",
+        cache_dir=model_args.cache_dir,
+        revision=model_args.model_revision,
+        use_auth_token=True if model_args.use_auth_token else None,
+    )
+
+    def model_init():
+        return AutoModelForImageClassification.from_pretrained(
+            model_args.model_name_or_path,
+            from_tf=bool(".ckpt" in model_args.model_name_or_path),
+            config=config,
+            cache_dir=model_args.cache_dir,
+            revision=model_args.model_revision,
+            use_auth_token=True if model_args.use_auth_token else None,
+        )
+
+    if data_args.max_train_samples is not None and data_args.max_train_samples < len(train_dataset):
+        indices = torch.randperm(data_args.max_train_samples).tolist()
+        train_dataset = torch.utils.data.Subset(train_dataset, indices)
+    if data_args.max_eval_samples is not None and data_args.max_eval_samples < len(eval_dataset):
+        indices = torch.randperm(data_args.max_eval_samples).tolist()
+        eval_dataset = torch.utils.data.Subset(eval_dataset, indices)
+    if data_args.max_predict_samples is not None and data_args.max_predict_samples < len(test_dataset):
+        indices = torch.randperm(data_args.max_predict_samples).tolist()
+        test_dataset = torch.utils.data.Subset(test_dataset, indices)
+
+    # NOTE - We aren't directly using this feature extractor since we defined custom transforms above.
```

Comment:
Explain a bit more why we are defining it anyway? It doesn't make sense otherwise.